src/basic/pidref.h

```diff
@@ -13,6 +13,11 @@ typedef struct PidRef {
 
 #define PIDREF_NULL (const PidRef) { .fd = -EBADF }
 
+/* A special pidref value that we are using when a PID shall be automatically acquired from some surrounding
+ * context, for example connection peer. Much like PIDREF_NULL it will be considerd unset by
+ * pidref_is_set().*/
+#define PIDREF_AUTOMATIC (const PidRef) { .pid = PID_AUTOMATIC, .fd = -EBADF }
```

Comment:
Hmm, PID_AUTOMATIC is defined in process-util.h, but not included here.

Reply:
yeah, otherwise we'd have a cyclic dep... this is on purpose

Reply:
Then, please move this and pidref_is_automatic() to process-util.h .
process-util.h already has several pidref_is_xyz(). Hence, should be ok.

Reply:
i instead fixed the headers now to allow the cyclic dep for cases like this, similar how we do this elesewhere